### PR TITLE
Include extension members on the extended type's page

### PR DESCRIPTION
**Pull request by srawlins**
In this feature, we list extension members of an extended type on that extended type's page. The members are listed inline with other instance members and are cross-linked over to the extension's member pages.

* Introduce `availableInstanceMethods` as a list combining both instance methods declared in a container, and instance methods declared in an applicable extension. The same for `availableInstanceOperators` and `availableInstanceFields`.
* Introduce some constructors like `Field.providedByExtension` for members that are provided by an extension.
* Add lots of documentation clarifying _where_ various members come from.
* We introduce a new getter on ElementType, `nameWithGenericsPlain`, which is not HTML-formatted, for use in HTML "title text."

Here are some screenshots:

Screenshot 2024-08-27 at 1 48 06 PM

---

Screenshot 2024-08-27 at 1 48 21 PM

---

Screenshot 2024-09-06 at 2 53 11 PM

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Comment by srawlins** [comment]
Be excited, @dart-lang/analyzer-team, this is ready for review 😁 😀 😁 😀

File: lib/src/generator/templates.aot_renderers_for_html.dart [hunk 1]
```diff
@@ -1497,20 +1497,31 @@ String renderSidebarForContainer<T extends Documentable>(
...
         buffer.write('''          <li''');
         if (context7.isInherited) {
           buffer.write(''' class="inherited"''');
         }
-        buffer.write('''>''');
+        buffer.write('''>
+            ''');
         buffer.write(context7.linkedName);
-        buffer.write('''</li>''');
+        if (context7.isProvidedByExtension) {
```
Comment (bwilkerson) [inline lib/src/generator/templates.aot_renderers_for_html.dart:1510]: This `if` statement appears to be duplicated several times in this file. I have to wonder whether we might be able to extract it into a utility method.

**Review by bwilkerson — APPROVED** [review]

File: lib/src/generator/templates.aot_renderers_for_html.dart (hunk 1 above)
Comment (srawlins) [reply lib/src/generator/templates.aot_renderers_for_html.dart]: SG. I've added a mustache partial for this text.

Comment (srawlins) [reply lib/src/generator/templates.aot_renderers_for_html.dart:1511]: Oops, Mustachio is apparently unable to handle this deduplication - an issue with referencing one partial multiple times from the same template, but each with different context types. I'm writing a unit test and I'll file a bug. For now I've left a TODO.

**Pull request merged**